datafusion/physical-plan/src/sorts/partial_sort.rs

```diff
@@ -563,7 +507,7 @@ mod tests {
                 "| 1 | 3 | 0 |",
                 "+---+---+---+",
             ];
-            assert_eq!(2, result.len());
+            assert_eq!(1, result.len());
```

Comment:
I think the sort in ExternalSort will merge the two `RecordBatch` to one if no spilling?